### ISSUE TITLE
Feature request: add flag to look for TOC content before `<!-- START -->`

### ISSUE THREAD

**Issue by haysclark** [thlorenz/doctoc#213]
I've been converting some docs to MD that have TOC content before the TOC itself. e.g. `## Preface`. Currently,  only generates TOC content for elements found after the ```<!-- START -->```. It would be nice if the  was a _doctoc_ flag would override this behavior and parse the entire document. 

Naturally, this might require adding additional features to exclude referencing the TOC itself, like IGNORE flags.:

```<!-- TOC-IGNORE-START -->```
```<!-- TOC-IGNORE-END -->```

**Comment by fab1o** [comment]
Is there a way to ignore sections? For instance, I want to ingnore Section 3, I have this readme:

# Title




## Section 1

## Section 2

## Section 3

**Comment by thlorenz** [comment]
None of the above two is possible ATM, but I'm open to a PR that makes that configurable.
`TOC-IGNORE-START` is a bit too specific, but something along the lines of `DOCTOC_INCLUDE_START` and `DOCTOC_INCLUDE_END` and `DOCTOC_SKIP_START/END` comments could work.

Best to discuss details here with some more detailed examples before jumping into the implementation.

**Comment by AndrewSouthpaw** [comment]
Agreed. It seems like the `--include-all` would be easier to pull off than the one about skippable headers. Probably want to treat them as a separate features. :)

**Comment by thompson-tomo** [comment]
The include all functionality has been available for a while with exclude functionality available via #359
